cmd/dra-example-kubeletplugin/driver.go

```diff
@@ -97,7 +98,21 @@ func (d *driver) nodePrepareResource(ctx context.Context, claim *drapbv1.Claim)
 		}
 	}
 
-	prepared, err := d.state.Prepare(resourceClaim)
+	deviceClasses := make(map[string]*resourceapi.DeviceClass)
+	for _, request := range resourceClaim.Spec.Devices.Requests {
+		deviceClass, err := d.client.ResourceV1alpha3().DeviceClasses().Get(
+			ctx,
+			request.DeviceClassName,
+			metav1.GetOptions{})
+		if err != nil {
+			return &drapbv1.NodePrepareResourceResponse{
+				Error: fmt.Sprintf("failed to fetch DeviceClass %s", request.DeviceClassName),
```

Comment:
Device classes may get deleted after allocation is complete. A driver should only depend on the claim status, but not on the device classes.

Reply:
Will update in nexgt revision based on answer to https://github.com/kubernetes-sigs/dra-example-driver/pull/54#discussion_r1732789769

Reply:
Updated.